src/darwin/Framework/CHIP/MTRCommissionableBrowser.mm

```diff
@@ -80,13 +86,23 @@ CHIP_ERROR Start(id<MTRCommissionableBrowserDelegate> delegate, dispatch_queue_t
 
     CHIP_ERROR Stop()
     {
+        assertChipStackLockedByCurrentThread();
+
         VerifyOrReturnError(mDelegate != nil, CHIP_ERROR_INCORRECT_STATE);
+        VerifyOrReturnError(mController != nil, CHIP_ERROR_INCORRECT_STATE);
         VerifyOrReturnError(mDispatchQueue != nil, CHIP_ERROR_INCORRECT_STATE);
         VerifyOrReturnError(mDiscoveredResults != nil, CHIP_ERROR_INCORRECT_STATE);
 
         mDelegate = nil;
+        mController = nil;
         mDispatchQueue = nil;
-        mDiscoveredResults = nil;
+
+        // Remove discovered ble devices instantly.
```

Comment:
Please document _why_ that's done.

Reply:
I have updated the code such that everything is removed.